docs/user-guide/kubeconfig-file.md

```diff
@@ -94,69 +110,58 @@ users:
-      1.  command line argument - `user` for user name and `cluster` for cluster name
-      1.  If context is present, then use the context's value
-      1.  Empty is allowed
-  1.  Determine the actual cluster info to use.  At this point, we may or may not have a cluster info.  Build each piece of the cluster info based on the chain (first hit wins):
-      1.  command line arguments - `server`, `api-version`, `certificate-authority`, and `insecure-skip-tls-verify`
-      1.  If cluster info is present and a value for the attribute is present, use it.
-      1.  If you don't have a server location, error.
-  1.  Determine the actual user info to use. User is built using the same rules as cluster info, EXCEPT that you can only have one authentication technique per user.
-      1. Load precedence is 1) command line flag, 2) user fields from kubeconfig
-      1. The command line flags are: `client-certificate`, `client-key`, `username`, `password`, and `token`.
-      1. If there are two conflicting techniques, fail.
-  1.  For any information still missing, use default values and potentially prompt for authentication information
-
-## Manipulation of kubeconfig via `kubectl config <subcommand>`
-
-In order to more easily manipulate kubeconfig files, there are a series of subcommands to `kubectl config` to help.
-See [kubectl/kubectl_config.md](kubectl/kubectl_config.md) for help.
-
-### Example
+Your kubeconfig files get loaded and merged based on the commands that you run from the terminal and the information specified in each kubeconfig file.
+
+The following loading and merging rules are ordered by their priority:
+
+ 1. Retrieve the kubeconfig files from disk using the following hierarchy and merge rules:
+   1. If the `CommandLineLocation` (the value of the `kubeconfig` command line option) is set, then use this file only and do not merge any other kubeconfig files. Only one instance of this flag is allowed in your kubeconfig files.
+   1. If `EnvVarLocation` (the value of $KUBECONFIG) is available, then use it to define all of the files that will be merged.
+   1. Merge the files using the following rules:  
+      * Empty filenames are ignored. Note: Files with non-deserializable content cause errors.
+      * The first time that a particular value or map key is found, then use that value or map key and ignore any subsequent occurrences.  
+        For example, if a file sets `CurrentContext` for the first time, then the context in that file is preserved and any other files that also set `CurrentContext` are ignored. Another example is if two files specify a `red-user`, then only the values from the first file's red-user are used. Even non-conflicting entries from the second file's red-user are ignored and not merged into the final kubeconfig file.
```

Comment:
Here we use 'CurrentContext' but in the example kubeconfig example its 'current-context', should this be changed or do we need to add more info here to clarify where 'CurrentContext' comes from?